daliuge-translator/dlg/dropmake/web/lg_web.py

```diff
@@ -197,6 +198,40 @@ def pgtjsonbody_get():
         response.status = 404
         return "{0}: JSON graph {1} not found\n".format(err_prefix, pgt_name)
 
+@get("/pgt_cwl")
+def pgtcwl_get():
+    """
+    Return CWL representation of the logical graph
+    """
+    pgt_name = request.query.get("pgt_name")
+
+    if pgt_exists(pgt_name):
+        # get PGT from manager
+        pgtp = pg_mgr.get_pgt(pgt_name)
+
+        # build node list
+        mgr_client = SimpleManagerClient()
+        node_list = mgr_client.nodes()
+
+        # mapping PGTP to resources (node list)
+        pg_spec = pgtp.to_pg_spec(node_list, ret_str=False)
```

Comment:
Note that cwl translation doesn't really require a *partitioned* physical graph (that's only required by daliuge). This could thus be simplified with `pgtp` being fed directly into `create_workflow`, making `SimpleManagerClient` redundant.

Reply:
Done.